examples/anchors.yaml

```diff
@@ -8,35 +8,36 @@ exporters:
     certificate: /app/cert.pem
     client_key: /app/cert.pem
     client_certificate: /app/cert.pem
-    headers:
-      api-key: !!str 1234
+    headers: "api-key=1234"
```

Comment:
I have some concerns with this.

Will it be possible later to not have the value appear in the config.yaml / environment variables content, and have a yaml node that represents a secret instead ?

Reply:
Sorry I'm not grasping the question. Can you elaborate?

Reply:
When some fields are secrets, like a password, or here an auth token:

```
headers:
  foo: 12
  bar: 24
  api-key: actual_auth_token <-- visible in plain view
```

The model can be changed to separate key names and key values:

```
headers:
  - key:
    name: foo
    value: 12
  - key:
    name: bar
    value: 24
  - key:
    name: api-key
    value: actual_auth_token <-- still visible in plain view
```

and then, a key value can be represented by an alternative node, which is a named proxy:

```
headers:
  - key:
    name: foo
    value: 12
  - key:
    name: bar
    value: 24
  - key:
    name: api-key
    secret:
      secret_name: "vendor_auth_token"
```

The secret name, "vendor_auth_token", is just a name and not the actual value.
The actual value (actual_auth_token) is retrieved from a secure location, referenced by name (vendor_auth_token).

The concern is that serializing key/value pairs in a comma delimited string prevents this.

Reply:
Ah I see. And that syntax is what is used in kubernetes, so there's definitely (popular) prior art for this type of syntax.

A counter argument is that `OTEL_EXPORTER_OTLP_HEADERS` has been around for a while and I haven't heard many (any?) issues with the fact that its a comma separated list of key value pairs. This design is essentially saying that because a comma separated list was "good enough" for `OTEL_EXPORTER_OTLP_HEADERS`, that its probably good enough for file configuration too. The trade off of course is expressiveness vs. ease of overriding w/ environment variables. The syntax you give above is highly expressive but not easily overridden with environment variables. Treating headers as a flattened comma separated list is not as expressive but is rally easy to override with environment variables for _most_ cases.

Reply:
In a world pre `config.yaml`, the application is configuring the SDK by writing explicit code.

When doing so, some configuration can be done using environment variables as a help, or ignoring environment variables and doing everything programmatically, or both.

Disclosure: For a major app at work, not a single environment variable is used, everything is done programmatically, finding the configuration data "elsewhere".

Now, in a setup using `config.yaml`, the main advantage of unifying all the configuration in a yaml document is that the entire configuration can change, without having to change a single line of code in the application.

Having auth tokens in a key/value delimited string in an environment variable (for people willing to use this), or in code (for people not willing to use this) was workable before, because the app code was ad hoc anyway.

In a world where all is unified with a yaml config file, parts which stands out for specific reasons (as in, I would like all the config in yaml ... **_except_** the sensitive part) are more likely to cause friction and stand out, compared to a pre config.yaml usage. What was good enough before may not be good enough now.

If there is a choice to make between config.yaml and security, it is already made: config.yaml will not be used, period.

So, all the work with configuration in general should have provisions to avoid these pitfalls.

As a side note, using key/value pairs in comma delimited strings (this PR) also broke the use case involving variables substitution:

```
headers:
  foo: 12
  bar: 34
  api-key: ${MY_VENDOR_TOKEN}
```

This setup is no longer working.